src/main/java/no/sikt/nva/pubchannels/handler/fetch/FetchByIdentifierAndYearHandler.java

```diff
@@ -12,41 +12,49 @@
 import java.net.URI;
 import java.time.Year;
 import java.util.List;
+import java.util.function.Function;
 import no.sikt.nva.pubchannels.channelregistry.ChannelRegistryClient;
+import no.sikt.nva.pubchannels.channelregistry.ChannelType;
+import no.sikt.nva.pubchannels.channelregistry.PublicationChannelMovedException;
+import no.sikt.nva.pubchannels.channelregistrycache.db.service.CacheService;
 import no.sikt.nva.pubchannels.handler.PublicationChannelClient;
+import no.sikt.nva.pubchannels.handler.ThirdPartyPublicationChannel;
 import nva.commons.apigateway.ApiGatewayHandler;
 import nva.commons.apigateway.RequestInfo;
 import nva.commons.apigateway.exceptions.ApiGatewayException;
 import nva.commons.apigateway.exceptions.BadRequestException;
 import nva.commons.core.Environment;
 import nva.commons.core.JacocoGenerated;
+import nva.commons.core.attempt.Failure;
 import nva.commons.core.paths.UriWrapper;
-import software.amazon.awssdk.services.s3.S3Client;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public abstract class FetchByIdentifierAndYearHandler<I, O> extends ApiGatewayHandler<I, O> {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(FetchByIdentifierAndYearHandler.class);
     private static final String ENV_API_DOMAIN = "API_DOMAIN";
     private static final String ENV_CUSTOM_DOMAIN_BASE_PATH = "CUSTOM_DOMAIN_BASE_PATH";
     private static final String YEAR_PATH_PARAM_NAME = "year";
     private static final String IDENTIFIER_PATH_PARAM_NAME = "identifier";
 
     protected final PublicationChannelClient publicationChannelClient;
-    protected final S3Client s3Client;
+    protected final CacheService cacheService;
     protected final boolean shouldUseCache;
 
     @JacocoGenerated
     protected FetchByIdentifierAndYearHandler(Class<I> iclass, Environment environment) {
         super(iclass, environment);
         this.publicationChannelClient = ChannelRegistryClient.defaultInstance();
-        this.s3Client = S3Client.create();
+        this.cacheService = CacheService.defaultInstance();
         this.shouldUseCache = Boolean.parseBoolean(environment.readEnv("SHOULD_USE_CACHE"));
     }
 
     protected FetchByIdentifierAndYearHandler(Class<I> requestClass, Environment environment,
-                                              PublicationChannelClient client, S3Client s3Client) {
+                                              PublicationChannelClient client, CacheService cacheService) {
```

Comment:
Response using cache only is as fast as api responses other places, under 200ms.